data-prep-pipeline/scripts/subtype_tree/create_subtype_tree.py

```diff
@@ -0,0 +1,619 @@
+    break any assumptions/rules that are used by the portal
+
+    Inputs:
+        - subtype_tree (pandas df): The final result of the subtype tree
+
+    Outputs: None (function will raise an error if any assert fails)
+    '''
+    
+    #assert that depmap codes are unique
+    assert(all(subtype_tree.groupby('DepmapModelType').size() == 1))
+
+    #assert that molecular codes are unique
+    assert(all(subtype_tree.groupby('MolecularSubtypeCode').size() == 1))
+
+    #assert there's no overlap between depmap and molecular codes
+    assert(len(
+        set.intersection(
+            set(subtype_tree.DepmapModelType.dropna().unique()),
+            set(subtype_tree.MolecularSubtypeCode.dropna().unique())    
+        )
+    ) == 0)
+
+    #assert that node names are unique
+    assert(all(subtype_tree.groupby('NodeName').size() == 1))
+
+
+### MAIN FUNCTION ###
+def create_subtype_tree(
+        model_taiga_id,
+        oncotree_taiga_id,
+        molecular_subtypes_taiga_id,
+        genetic_subtypes_whitelist_folder,
+        genetic_subtypes_whitelist_filename
+    ):
+    models, oncotree, genetic_subtypes, gs_whitelist = load_data(
+        model_taiga_id,
+        oncotree_taiga_id,
+        molecular_subtypes_taiga_id,
+        genetic_subtypes_whitelist_folder,
+        genetic_subtypes_whitelist_filename
+    )
+
+    oncotable = create_oncotable(oncotree)
+
+    oncotable_plus = add_depmap_nodes(models, oncotable)
+
+    subtype_tree_gs = add_disease_restricted_genetic_subtypes(gs_whitelist, oncotable_plus)
+
+    molecular_subtype_tree = create_molecular_subtype_tree(genetic_subtypes)
+
+    all_subtypes = pd.concat([
+        subtype_tree_gs.assign(TreeType = 'Lineage'),
+        molecular_subtype_tree.assign(TreeType = 'MolecularSubtype')
+    ])
+
+    subtype_tree = create_subtype_tree_with_names(all_subtypes)
+
+    # Verify results
+    sanity_check_results(subtype_tree)
+
+    return subtype_tree
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Create SubtypeTree"
+    )
+    parser.add_argument("model", help="Taiga ID of model table")
+    parser.add_argument("oncotree", help="Taiga ID of oncotree")
+    parser.add_argument("molecular_subtypes", help="Taiga ID of Omics Inferred Molecular Subtypes")
+    parser.add_argument("genetic_subtypes_whitelist_folder", help="Taiga folder of lineage-based genetic subtype whitelist")
+    parser.add_argument("genetic_subtypes_whitelist_filename", help="Taiga filename of lineage-based genetic subtype whitelist")
+    parser.add_argument("output", help="filepath to write the output")
+    args = parser.parse_args()
+
+    subtype_tree = create_subtype_tree(
+        args.model,
+        args.oncotree,
+        args.molecular_subtypes,
+        args.genetic_subtypes_whitelist_folder,
+        args.genetic_subtypes_whitelist_filename
+    )
+
+    if subtype_tree is not None:
+        subtype_tree.to_csv(args.output)
```

Comment:
Like above, I wonder if we should take out this None check. Maybe I'm misunderstanding how this works.